cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -126,6 +131,7 @@ public void startPreview(PreviewRequest<?> previewRequest) throws Exception {
                                             artifactId.toId(), config, NOOP_PROGRAM_TERMINATOR);
     } catch (Exception e) {
       this.status = new PreviewStatus(PreviewStatus.Status.DEPLOY_FAILED, new BasicThrowable(e));
+      shutDownUnrequiredServices();
```

Comment:
No need to call it here. Caller(who starts preview run) should stop PreviewRunner by catching this exception.